examples/pipelines/finetune_stable_diffusion/components/dataset_loader_component/src/main.py

```diff
@@ -1,169 +1,66 @@
-                                            id_prefix=namespace)
-        # Write parquet dataset
-        parquet_helpers.write_dataset_parquet \
-            (dataset_parquet_path=dataset_parquet_tmp_path,
-             data_iterable_producer=storage_helpers.get_blob_metadata,
-             storage_client=storage_client,
-             bucket_name=source_dataset_bucket,
-             prefix=source_dataset_blob,
-             id_prefix=namespace)
-
-        logger.info('Parquet manifest files updated')
-
-        # Upload the parquet
-        storage_helpers.upload_file_to_bucket(storage_client=storage_client,
-                                              file_to_upload_path=dataset_parquet_tmp_path,
-                                              bucket_name=artifact_bucket,
-                                              blob_path=dataset_parquet_blob_path)
-        storage_helpers.upload_file_to_bucket(storage_client=storage_client,
-                                              file_to_upload_path=index_parquet_tmp_path,
-                                              bucket_name=artifact_bucket,
-                                              blob_path=index_parquet_blob_path)
-        logger.info('Parquet manifest files uploaded to GCS')
-
-        data_manifest = DataManifest()
-        data_manifest.dataset_id = f"{run_id}_{component_name}"
-        data_manifest.index = f'gs://{artifact_bucket}/{index_parquet_blob_path}'
-        data_manifest.associated_data.dataset[namespace] = \
-            f'gs://{artifact_bucket}/{dataset_parquet_blob_path}'
-        data_manifest.metadata.branch = ""  # TODO: Fill from docker build env var
-        data_manifest.metadata.commit_hash = ""  # TODO: Fill from docker build env var
-        data_manifest.metadata.creation_date = datetime.now().strftime("%d-%m-%Y_%H-%M-%S")
-        data_manifest.metadata.run_id = run_id
-
-        logger.info('Manifest file created and updated')
-
-        # Write manifest to outputPath
-        Path(data_manifest_path).parent.mkdir(parents=True, exist_ok=True)
-        Path(data_manifest_path).write_text(data_manifest.to_json())
-
-        logger.info('Manifest file written to %s', data_manifest_path)
-
-        # Clean up temporary storage
-        logger.info('Files removed from temporary storage.')
-        logger.info('Job completed.')
+from typing import Optional, Union, Dict
+
+from datasets import Dataset, load_dataset
+
+from express.components.hf_datasets_components import HFDatasetsLoaderComponent, HFDatasetsDatasetDraft
+from express.logger import configure_logging
+
+configure_logging()
+logger = logging.getLogger(__name__)
+
+
+def create_metadata(examples):
+    images = examples["image"]
+    sizes = [image.size for image in images]
+    
+    batch = {}
+    batch["width"] = [size[0] for size in sizes]
+    batch["height"] = [size[1] for size in sizes]
+    
+    return batch
+
+
+class DatasetLoaderComponent(HFDatasetsLoaderComponent):
```

Comment:
Maybe this should be changed to `hub_dataset_loader` or something else to distinguish it from loading from cloud + update doc strings